pkgs/development/compilers/gcc/4.8/default.nix

```diff
@@ -359,6 +359,8 @@ stdenv.mkDerivation ({
     }
     ${if (stdenv ? glibc && cross == null)
       then " --with-native-system-header-dir=${stdenv.glibc.dev}/include"
+      else if (stdenv ? libc && cross == null)
```

Comment:
You should be able to change the existing check to
````
     ${if cross == null
￼        then " --with-native-system-header-dir=${lib.getDev stdenv.libc}/include"
````

which should do the right thing for all platforms.